launcher/internal/gpu/driverinstaller.go

```diff
@@ -168,15 +180,31 @@ func (di *DriverInstaller) InstallGPUDrivers(ctx context.Context) error {
 	return nil
 }
 
-func getInstallerImageReference() (string, error) {
-	installerImageRefBytes, err := exec.Command("cos-extensions", "list", "--", "--gpu-installer").Output()
+func getInstallerImageReference(installerImageRefFile string) (string, error) {
+	imageRefBytes, err := os.ReadFile(installerImageRefFile)
 	if err != nil {
 		return "", fmt.Errorf("failed to get the cos-gpu-installer version: %v", err)
 	}
-	installerImageRef := strings.TrimSpace(string(installerImageRefBytes))
+	installerImageRef := strings.TrimSpace(string(imageRefBytes))
+	if len(installerImageRef) < 1 {
+		return "", fmt.Errorf("empty value of cos-gpu-installer image reference")
+	}
 	return installerImageRef, nil
 }
 
+func verifyInstallerImageDigest(image containerd.Image, referenceDigestFile string) error {
+	installerDigest := image.Target().Digest.String()
+	imageDigestBytes, err := os.ReadFile(referenceDigestFile)
+	if err != nil {
+		return fmt.Errorf("failed to get the cos-gpu-installer image digest: %v", err)
+	}
+	expectedInstallerDigest := strings.TrimSpace(string(imageDigestBytes))
+	if installerDigest == expectedInstallerDigest {
+		return nil
+	}
+	return fmt.Errorf("cos_gpu_installer image digest verification failed - expected : %s, actual : %s", expectedInstallerDigest, installerDigest)
```

Comment:
nit: it's more typical to invert the error check (installerDigest != expectedInstallerDigest) and return nil at the end.